libs/sdk-ui-ext/src/internal/components/pluggableVisualizations/drillDownUtil.ts

```diff
@@ -0,0 +1,165 @@
+}
+
+type TColumnWidths = {
+    attributeColumnWidthItem?: {
+        attributeIdentifier: string;
+        width: {
+            value: number;
+        };
+    };
+    measureColumnWidthItem?: {
+        width: {
+            value: number;
+        };
+    };
+};
+
+enum ENUM_PROPERTIES_TYPE {
+    CONTROLS = "controls",
+}
+
+export function modifyBucketsAttributesForDrillDown(
+    insight: IInsight,
+    drillDefinition: IImplicitDrillDown,
+): IInsight {
+    const removedLeftAttributes = insightReduceItems(
+        insight,
+        (acc: IAttributeOrMeasure[], cur: IAttributeOrMeasure): IAttributeOrMeasure[] => {
+            if (isAttribute(cur) && matchesDrillDownTargetAttribute(drillDefinition, cur)) {
+                return [cur];
+            }
+
+            return [...acc, cur];
+        },
+    );
+
+    return insightModifyItems(
+        removedLeftAttributes,
+        (bucketItem: IAttributeOrMeasure): IAttributeOrMeasure => {
+            if (isAttribute(bucketItem) && matchesDrillDownTargetAttribute(drillDefinition, bucketItem)) {
+                const displayForm = drillDownDisplayForm(drillDefinition);
+                return modifyAttribute(bucketItem, (a) => a.displayForm(displayForm).noAlias());
+            }
+            return bucketItem;
+        },
+    );
+}
+
+function removePropertiesForRemovedAttributes(insight: IInsight): IInsight {
+    const properties: VisualizationProperties = insightProperties(insight);
+
+    if (!properties) {
+        return insight;
+    }
+
+    const identifiers = insightItems(insight).map((bucketItem: IAttributeOrMeasure) =>
+        bucketItemLocalId(bucketItem),
+    );
+
+    const result = Object.entries(properties).reduce((acc, [key, value]) => {
+        if (key === ENUM_PROPERTIES_TYPE.CONTROLS && value.columnWidths) {
+            const columns = value.columnWidths.filter(
+                (c: TColumnWidths) =>
+                    c?.attributeColumnWidthItem?.attributeIdentifier === undefined ||
+                    identifiers.includes(c.attributeColumnWidthItem.attributeIdentifier),
+            );
+
+            return {
+                ...acc,
+                [key]: {
+                    columnWidths: columns,
+                },
+            };
+        }
+
+        return { ...acc };
+    }, properties);
+
+    return insightSetProperties(insight, result);
+}
+
+export function sanitizeTableProperties(insight: IInsight): IInsight {
+    return removePropertiesForRemovedAttributes(insight);
+}
+
+export function convertIntersectionToFilters(intersections: IDrillEventIntersectionElement[]): IFilter[] {
+    return intersections
+        .map((intersection) => intersection.header)
+        .filter(isDrillIntersectionAttributeItem)
+        .map((header) =>
+            newPositiveAttributeFilter(uriRef(header.attributeHeader.uri), {
+                uris: [header.attributeHeaderItem.uri],
+            }),
```

Comment:
we cannot hardcode uriRefs -> tiger incompatible. i realize we have a small hole in the headers where we do not include the opaque `ref` which this code would just take and use as-is. alas, we need to correct it as part of this.

Reply:
Thanks for augmenting, hopefully this https://github.com/gooddata/gooddata-ui-sdk/pull/585/commits/f22ecc9ba50182de2b0c481f512cf1ffbe1a3b28 and this https://github.com/gooddata/gooddata-ui-sdk/pull/585/commits/b414818b4ab5ffe2fda4e53f48e2c465d18825b3 would do :-)